spec.md

```diff
@@ -477,7 +477,9 @@ integer literals.  Any use in place of an integer is disallowed.
 # Circuit Components
 
 Circuit components are the named objects which may be contained within a module.
-They are: nodes, wires, registers, submodule instances, and ports.
+They are: nodes, wires, registers, ports, and submodule instances.
+
+TODO: What about `mem`\s?
```

Comment:
Have I captured all the circuit components here? 

I vaguely remember thinking there were 7 for some reason back in April. I know I counted `output` and `input` as separate, but that leaves me feeling like I'm missing one. Maybe I had included `mem`.

Reply:
It looks like `mem` is a circuit component. Added a section for it.